arch/sim/src/Makefile

```diff
@@ -285,14 +285,14 @@ nuttx-names.dat: nuttx-names.in
 nuttx$(EXEEXT): libarch$(LIBEXT) board/libboard$(LIBEXT) $(HEADOBJ) $(LINKOBJS) $(HOSTOBJS) nuttx-names.dat
 	$(Q) echo "LD:  nuttx$(EXEEXT)"
 	$(Q) $(LD) -r $(LDLINKFLAGS) $(RELPATHS) $(EXTRA_LIBPATHS) -o nuttx.rel $(REQUIREDOBJS) $(LDSTARTGROUP) $(RELLIBS) $(EXTRA_LIBS) $(LDENDGROUP)
-ifeq ("$(shell $(CC) --version | grep clang)","")
-   # none clang based native compilers need opjcopy to build simulation
+ifneq ($(CONFIG_HOST_MACOS),y)
```

Comment:
should we skip init_array hack on macOS?

Reply:
we can. but it's a separate change i guess.

Reply:
Why the separate change.

Reply:
fixing the build and optimizing it by skipping unnecessary steps are two different things.

also, the current version was tested and i don't think i will test the "optimized" version today.